element_array_ephys/probe.py

```diff
@@ -96,60 +96,26 @@ def create_neuropixels_probe(probe_type: str = "neuropixels 1.0 - 3A"):
 
         For electrode location, the (0, 0) is the
          bottom left corner of the probe (ignore the tip portion)
-        Electrode numbering is 1-indexing
+        Electrode numbering is 0-indexing
         """
 
-        neuropixels_probes_config = {
-            "neuropixels 1.0 - 3A": dict(
-                site_count_per_shank=960,
-                col_spacing=32,
-                row_spacing=20,
-                white_spacing=16,
-                col_count_per_shank=2,
-                shank_count=1,
-                shank_spacing=0,
-            ),
-            "neuropixels 1.0 - 3B": dict(
-                site_count_per_shank=960,
-                col_spacing=32,
-                row_spacing=20,
-                white_spacing=16,
-                col_count_per_shank=2,
-                shank_count=1,
-                shank_spacing=0,
-            ),
-            "neuropixels UHD": dict(
-                site_count_per_shank=384,
-                col_spacing=6,
-                row_spacing=6,
-                white_spacing=0,
-                col_count_per_shank=8,
-                shank_count=1,
-                shank_spacing=0,
-            ),
-            "neuropixels 2.0 - SS": dict(
-                site_count_per_shank=1280,
-                col_spacing=32,
-                row_spacing=15,
-                white_spacing=0,
-                col_count_per_shank=2,
-                shank_count=1,
-                shank_spacing=250,
-            ),
-            "neuropixels 2.0 - MS": dict(
-                site_count_per_shank=1280,
-                col_spacing=32,
-                row_spacing=15,
-                white_spacing=0,
-                col_count_per_shank=2,
-                shank_count=4,
-                shank_spacing=250,
-            ),
-        }
+        npx_probes_config = probe_geometry.M
+        npx_probes_config["neuropixels 1.0 - 3A"] = npx_probes_config["3A"]
+        npx_probes_config["neuropixels 1.0 - 3B"] = npx_probes_config["NP1010"]
+        npx_probes_config["neuropixels UHD"] = npx_probes_config["NP1100"]
+        npx_probes_config["neuropixels 2.0 - SS"] = npx_probes_config["NP2000"]
+        npx_probes_config["neuropixels 2.0 - MS"] = npx_probes_config["NP2010"]
 
         probe_type = {"probe_type": probe_type}
-        electrode_layouts = build_electrode_layouts(
-            **{**neuropixels_probes_config[probe_type["probe_type"]], **probe_type}
+        probe_params = {
+            n: v
+            for n, v in zip(
+                probe_geometry.geom_param_names,
+                npx_probes_config[probe_type["probe_type"]],
+            )
+        }
+        electrode_layouts = probe_geometry.build_npx_probe(
+            **{**probe_params, **probe_type}
```

Comment:
```suggestion
        electrode_layouts = probe_geometry.build_npx_probe(
            **probe_params, **probe_type
```

Reply:
The suggested change would error out if there are duplicating keys between the two dictionaries. 
The original approach is more robust in the sense that it would merge dictionaries.